bank/services/jpmorgan/jpmorgan.go

```diff
@@ -0,0 +1,232 @@
+}
+
+func (b bankJPMorgan) ProcessBoleto(boleto *models.BoletoRequest) (models.BoletoResponse, error) {
+	errs := b.ValidateBoleto(boleto)
+	if len(errs) > 0 {
+		return models.BoletoResponse{Errors: errs}, nil
+	}
+
+	return b.RegisterBoleto(boleto)
+}
+
+func (b bankJPMorgan) ValidateBoleto(request *models.BoletoRequest) models.Errors {
+	return models.Errors(b.validate.Assert(request))
+}
+
+func (b bankJPMorgan) GetBankNumber() models.BankNumber {
+	return models.JPMorgan
+}
+
+func (b bankJPMorgan) GetBankNameIntegration() string {
+	return "JPMorgan"
+}
+
+func (b bankJPMorgan) GetErrorsMap() map[string]int {
+	var erros = map[string]int{
+		"BOL-5":                          http.StatusBadRequest,
+		"BOL-7":                          http.StatusBadRequest,
+		"BOL-144":                        http.StatusBadRequest,
+		"GCA-001":                        http.StatusBadGateway,
+		"GCA-111":                        http.StatusBadRequest,
+		"GCA-003":                        http.StatusBadGateway,
+		"BOL-2":                          http.StatusBadRequest,
+		"BOL-3":                          http.StatusBadRequest,
+		"BOL-4":                          http.StatusBadRequest,
+		"BOL-1":                          http.StatusBadGateway,
+		"Signature Verification Failure": http.StatusInternalServerError,
+		"Authentication Failure":         http.StatusInternalServerError,
+		"Internal Error - Contact Service Provider": http.StatusBadGateway,
+	}
+	return erros
+}
+
+func (b bankJPMorgan) Log() *log.Log {
+	return b.log
+}
+
+func (b bankJPMorgan) RegisterBoleto(boleto *models.BoletoRequest) (models.BoletoResponse, error) {
+	var response string
+	var respHeader map[string]interface{}
+	var status int
+	var err error
+	var bodyEncripted string
+
+	JPMorganURL := config.Get().URLJPMorgan
+	boleto.Title.BoletoType, boleto.Title.BoletoTypeCode = getBoletoType(boleto)
+
+	body := flow.NewFlow().From("message://?source=inline", boleto, templateRequest, tmpl.GetFuncMaps()).GetBody().(string)
+	head := hearders()
+
+	bodyEncripted, BoletoResponseError := b.encriptedBody(body)
+	if BoletoResponseError != nil {
+		return *BoletoResponseError, nil
+	}
+
+	b.log.Request(body, JPMorganURL, getLogRequestProperties(head, bodyEncripted))
+
+	duration := util.Duration(func() {
+		if config.Get().MockMode {
+			response, respHeader, status, err = util.PostWithHeader(JPMorganURL, body, config.Get().TimeoutDefault, head)
+		} else {
+			response, respHeader, status, err = util.PostTLSWithHeader(JPMorganURL, bodyEncripted, config.Get().TimeoutDefault, head, b.transport)
+		}
+
+	})
+	metrics.PushTimingMetric("jpmorgan-register-boleto-time", duration.Seconds())
+
+	b.log.Response(response, JPMorganURL, nil)
+
+	return mapJPMorganResponse(boleto, getContentType(respHeader), response, status, err), nil
+}
+
+func (b bankJPMorgan) encriptedBody(body string) (string, *models.BoletoResponse) {
+	errorResponse := models.GetBoletoResponseError("MP500", "Internal Error")
```

Comment:
@davidsmedeiros Poderíamos colocar uma mensagem mais significativa sobre o erro aqui, ao invés do `Internal Error`, de forma a melhor identificar o problema ... foi mal não ter dado essa sugestão inicialmente 😢 
```suggestion
}

func (b bankJPMorgan) encryptedBody(body string) (string, *models.BoletoResponse) {
	sk, err := certificate.GetCertificateFromStore(config.Get().AzureStorageJPMorganSignCrtName)
	if err != nil {
		responseError := models.GetBoletoResponseError("MP500", "fail on load digital signature certificate")
		return "", &responseError
	}
	bodyEncripted, err := b.jwtSigner.Sign(body, sk.([]byte))
	if err != nil {
		responseError := models.GetBoletoResponseError("MP500", "fail on encrypt request body")
		return "", &responseError
	}
	return bodyEncripted, nil
}
```